engine/redis/engine.go

```diff
@@ -74,6 +84,24 @@ func (e *Engine) Publish(namespace, queue string, body []byte, ttlSecond, delayS
 	if tries == 0 {
 		return job.ID(), nil
 	}
+	if e.dataMgr != nil && delaySecond > DefaultPumpInTh {
```

Comment:
Can we take those codes into a single function to make the publish part clean?

Reply:
yes, will do